components/drafts/thread_draft/thread_draft.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useCallback, useMemo, useEffect} from 'react';
+import {useDispatch} from 'react-redux';
+
+import {getPost} from 'mattermost-redux/actions/posts';
+import {$ID} from 'mattermost-redux/types/utilities';
+import {UserThread, UserThreadSynthetic} from 'mattermost-redux/types/threads';
+import {Channel} from 'mattermost-redux/types/channels';
+import {Post} from 'mattermost-redux/types/posts';
+import {UserProfile, UserStatus} from 'mattermost-redux/types/users';
+
+import {selectPost} from 'actions/views/rhs';
+import {removeDraft} from 'actions/views/drafts';
+import {makeOnSubmit} from 'actions/views/create_comment';
+
+import {PostDraft} from 'types/store/rhs';
+
+import DraftTitle from '../draft_title';
+import DraftActions from '../draft_actions';
+import Panel from '../panel/panel';
+import Header from '../panel/panel_header';
+import Body from '../panel/panel_body';
+
+type Props = {
+    channel: Channel;
+    displayName: string;
+    draftId: string;
+    id: $ID<UserThread | UserThreadSynthetic>;
+    status: UserStatus['status'];
+    thread?: UserThread | UserThreadSynthetic;
+    type: 'channel' | 'thread';
+    user: UserProfile;
+    value: PostDraft;
+}
+
+function ThreadDraft({
+    channel,
+    displayName,
+    draftId,
+    id,
+    status,
+    thread,
+    type,
+    user,
+    value,
+}: Props) {
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        if (!thread?.id) {
+            dispatch(getPost(id));
```

Comment:
yep you are right, `getThread` requires team id, while `getPost`'s signature is quite simple, requires only the post id.
that's why I'm just getting the post, and then creating a synthetic thread, since we don't care about the thread per se, but just if the post exists or not.